### PR TITLE
Bugfix: BulkLoader_Result class - Deleted Items

### PR DESCRIPTION
**Pull request by AntonyThorpe**
The Deleted function, within BulkLoader_Result class, calls mapToArrayList, which tries to find the recently deleted dataobjects.
To correct:
- when calling addDeleted, store a map of each dataobject within the $this->deleted array; and,
- have the Deleted function return an ArrayList based upon the $this->deleted array.

### CODE REVIEW

**Comment by sminnee** [comment]
Thanks, @AntonyThorpe! Any chance you can write a test that covers this issue? Something that would fail without your change and passes with your change.

**Comment by AntonyThorpe** [comment]
Tests added as requested.  Unfortunately the Travis CI build failed due to a Behat CMS test (file1.jpg missing in the HTML following inserting from the CMS file store).  Can you please advise the next steps?  Sorry, I am not familiar with Behat tests.  Thank you.

File: dev/BulkLoader.php [hunk 1]
```diff
@@ -391,7 +391,7 @@ public function addUpdated($obj, $message = null) {
 	 */
 	public function addDeleted($obj, $message = null) {
 		$data = $obj->toMap();
-		$data['Message'] = $message;
+		$data['_BulkLoaderMessage'] = $message;
```
Comment (sminnee) [inline dev/BulkLoader.php]: Why did you change Message to _BuklLoaderMessage? It was Message in the original commit; was this a bug?

**Comment by sminnee** [comment]
I'll re-execute a couple of builds to see if that failure is related to this PR.

File: dev/BulkLoader.php (hunk 1 above)
Comment (sminnee) [inline dev/BulkLoader.php]: Oh, I see. It's mimicking the behaviour of `mapToArrayList()`

**Comment by dhensby** [comment]
Thanks for this!
